modin/pandas/series.py

```diff
@@ -1389,14 +1389,25 @@ def update(self, other):
     def value_counts(
         self, normalize=False, sort=True, ascending=False, bins=None, dropna=True
     ):
-        return self.__constructor__(
-            query_compiler=self._query_compiler.value_counts(
+        if bins is not None:
+            # Potentially we could implement `cut` function from Pandas API, which
+            # bins values into intervals, and then we can just count them as regular values.
+            # TODO: new_self = Series(pd.cut(self, bins, include_lowest=True), dtype="interval")
+            return self._default_to_pandas(
+                pandas.Series.value_counts,
                 normalize=normalize,
                 sort=sort,
                 ascending=ascending,
                 bins=bins,
                 dropna=dropna,
             )
+
+        return super(Series, self).value_counts(
```

Comment:
I prefer rewriting this into the API itself to reduce the size of the query compiler.

```python
result = self.groupby(by=self, sort=sort and not ascending, dropna=dropna).size()
```

Reply:
Handle `ascending` with special case because it is uncommon. The sort can be post-processing.